src/library/f3d_engine.h

```diff
@@ -19,14 +20,29 @@ class interactor;
 class engine
 {
 public:
-  enum class WindowTypeEnum
+  struct engine_exception : public std::exception
   {
-    WINDOW_NO_RENDER,
-    WINDOW_OFFSCREEN,
-    WINDOW_STANDARD
+  };
+  struct window_engine_exception : public engine_exception
+  {
+    const char* what() const throw() { return "Cannot create window with this engine"; }
+  };
+  struct interactor_engine_exception : public engine_exception
+  {
+    const char* what() const throw() { return "Cannot create interactor with this engine"; }
   };
 
-  engine(WindowTypeEnum windowType);
+  //======== Engine Flags =============
+  // engine::NO_WINDOW: Do not create a window to render into.
+  // engine::NO_INTERACTOR: Do not create an interactor to interact with
+  // engine::WINDOW_OFFSCREEN: Create an offscreen window to render into, ignored with NO_WINDOW
+  using engine_flags_t = std::bitset<8>;
+  static constexpr engine_flags_t FLAGS_NONE = engine_flags_t(0x0000);
+  static constexpr engine_flags_t NO_WINDOW = engine_flags_t(0x0001);
```

Comment:
Having `NO_*` flags look weird to me. What about doing the opposite? `WINDOW_NATIVE` and `INTERACTOR_DEFAULT` flags?

Reply:
done